File: leetcode/AIcoding/2.Gemini-2.0/cpp_solutions/1189-maximum-number-of-balloons.cpp
#include <string>
#include <algorithm>
#include <map>

using namespace std;

class Solution {
public:
    int maxNumberOfBalloons(string text) {
        map<char, int> textCount;
        for (char c : text) {
            textCount[c]++;
        }

        map<char, int> balloonCount = {
            {'b', 0},
            {'a', 0},
            {'l', 0},
            {'o', 0},
            {'n', 0}
        };

        balloonCount['b'] = textCount['b'];
        balloonCount['a'] = textCount['a'];
        balloonCount['l'] = textCount['l'] / 2;
        balloonCount['o'] = textCount['o'] / 2;
        balloonCount['n'] = textCount['n'];

        int minCount = INT_MAX;
        for (auto const& [key, val] : balloonCount) {
            minCount = min(minCount, val);
        }

        return minCount;
    }
};

File: leetcode/AIcoding/2.Gemini-2.0/javascript_solutions/1189-maximum-number-of-balloons.js
var maxNumberOfBalloons = function(text) {
    let count = {};
    for (let char of text) {
        count[char] = (count[char] || 0) + 1;
    }

    let balloonCount = 0;
    while (true) {
        if (count['b'] >= 1 && count['a'] >= 1 && count['l'] >= 2 && count['o'] >= 2 && count['n'] >= 1) {
            count['b']--;
            count['a']--;
            count['l'] -= 2;
            count['o'] -= 2;
            count['n']--;
            balloonCount++;
        } else {
            break;
        }
    }

    return balloonCount;
};